packages/cli/lib/base-generator.js

```diff
@@ -28,11 +31,218 @@ module.exports = class BaseGenerator extends Generator {
    * Subclasses can extend _setupGenerator() to set up the generator
    */
   _setupGenerator() {
+    this.option('config', {
+      type: String,
+      alias: 'c',
+      description: 'JSON file to configure options',
+    });
+
+    this.option('skip-optional-prompts', {
```

Comment:
I'm not sure how I feel about a such a long flag name from a UX perspective. Can we consider a simpler / easier to type / use / remember flag such as `--skip` or `--skip-optional` or `--express` (but may be confusing with Express framework) or `--quick` or `--defaults` ... anything smaller than the current flag.

Reply:
I was proposing `--non-interactive`, but that name has a slightly different meaning.

To be honest, I don't see much value in `--skip-default-prompts` unless it's used with the JSON mode. I am proposing to merge these two modes together and use the `--config` option to drive the behavior.

 - Answers are read from JSON - use non-interactive mode (skip default prompts) 
 - Answers are read from UI prompts - use interactive mode (ask all prompts)

That way there is only one new CLI option to add (`--config`).